perspective_reddit_bot/log_subreddit_comments.py

```diff
@@ -124,15 +124,19 @@ def log_subreddit(creds, subreddits, output_dir):
                        password=creds['reddit_password'])
   subreddit = reddit.subreddit(all_subs)
 
-  for i, comment in enumerate(comment_stream(subreddit.stream)):
+  while True:
     try:
-      print('.', end='')
-      sys.stdout.flush()
-      output_record = create_comment_output_record(comment)
-      if i % 25 == 0: print_comment(i, output_record)
-      append_records(output_path, [output_record])
+      for i, comment in enumerate(comment_stream(subreddit.stream)):
```

Comment:
Is there a chance this will skip comments if it throws away comments buffered by comment_stream? Or even duplicate comments because comment_stream goes back 100 comments historically? Maybe we can write a small test for these two cases?

Reply:
Yes, this error handling is definitely not great - I just added it because I noticed all the logging processes would quit when Reddit had some server-side issues. This is essentially like adding a `while true; do python log_subreddit_comments.py ...; done` retry loop around the whole program.

I'm pretty sure it _does_ cause duplicates (which is why I had to write #31 to dedup logs...). Now that I look at it again, I see if we make `seen_comment_ids` a global, we may be able to avoid dupes. Will do that.